### PR TITLE
Fix datetime serialization error on Windows

### LINKED ISSUE #349: DateTimeField serialization error on Windows 10
## Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* Operating System: Windows 10
* Python Version: 3.8.0
* aiogram version: 2.8
* aiohttp version: 3.6.2
* uvloop version (if installed): not installed

## Expected Behavior

Serialized DateTimeField

## Current Behavior

OSError: [Errno 22] Invalid argument

## Failure Information (for bugs)

When datetime is datetime.datetime(1970, 1, 1) the serialization on Windows 10 fails with OSError

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. `await bot.kick_user(chat_id, user_id)`
2. `print(await bot.get_chat_member(chat_id, user_id))`
3. you get it... no, you

### Failure Logs

```
Traceback (most recent call last):
  File "C:\Users\Admin\Desktop\projects\photo_city_bot\bot_api\endpoints\unban_users.py", line 36, in post
    print(chat_member)
  File "C:\Users\Admin\Desktop\projects\photo_city_bot\venv\lib\site-packages\aiogram\types\base.py", line 204, in __str__
    return self.as_json()
  File "C:\Users\Admin\Desktop\projects\photo_city_bot\venv\lib\site-packages\aiogram\types\base.py", line 192, in as_json
    return json.dumps(self.to_python())
  File "C:\Users\Admin\Desktop\projects\photo_city_bot\venv\lib\site-packages\aiogram\types\base.py", line 169, in to_python
    value = self.props[name].export(self)
  File "C:\Users\Admin\Desktop\projects\photo_city_bot\venv\lib\site-packages\aiogram\types\fields.py", line 95, in export
    return self.serialize(self.get_value(instance))
  File "C:\Users\Admin\Desktop\projects\photo_city_bot\venv\lib\site-packages\aiogram\types\fields.py", line 180, in serialize
    timestamp = value.timestamp()
OSError: [Errno 22] Invalid argument
```

I would fix it like that:
```
class DateTimeField(Field):
    """
    In this field st_ored datetime

    in: unixtime
    out: datetime
    """

    def serialize(self, value: datetime.datetime):
        print(value)
        if os.name == 'nt':
            timestamp = (value - datetime.datetime(1970, 1, 1)).total_seconds()
        else:
            timestamp = value.timestamp()

        return round(timestamp)

    def deserialize(self, value, parent=None):
        return datetime.datetime.fromtimestamp(value)
```
but I'm not sure if it is normal, doesn't seem to be ok to me xD

### PR DESCRIPTION
**Pull request by arynyklas**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

Fixes #349 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

```
@dp.chat_join_request_handler()
async def chat_join_request_handler(chat_join_request: types.ChatJoinRequest) ->None:
    print(chat_join_request)
```

**Test Configuration**:
* Operating System: Windows 10 Pro 21H1 (19043.1586)
* Python version: Python 3.10.4 (tags/v3.10.4:9d38120) [MSC v.1929 64 bit (AMD64)]

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: aiogram/types/fields.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ class DateTimeField(Field):
     """
 
     def serialize(self, value: datetime.datetime):
-        return round(value.timestamp())
+        return (value - datetime.datetime(1970, 1, 1)).total_seconds()
```
Comment (JrooTJunior) [inline aiogram/types/fields.py:172]: round returns int, total_seconds() returns float, so this methods in not equivalent
Also this dirty-hack is needed only on Windows, so please make an platform-specific function and use it here.

For example at module root (pseudo-code):
```python
if sys.platform == "win32":
    def _serialize_datetime(value):
        return round(...total_seconds())
else:
    def _serialize_datetime(value):
        return round(...timestamp())
```

**Review by JrooTJunior — CHANGES_REQUESTED** [review]

**Review by JrooTJunior — APPROVED** [review]

**Comment by fifthsound** [comment]
@arynyklas  For win7 have to change also deserialize:

`    
if sys.platform == "win32":
      def serialize(self, value: datetime.datetime):
          return round((value - datetime.datetime(1970, 1, 1)).total_seconds())
      **def deserialize(self, value, parent=None):
          return value**
  else:
      def serialize(self, value: datetime.datetime):
          return round(value.timestamp())
      def deserialize(self, value, parent=None):
          return datetime.datetime.fromtimestamp(value)
`

**Comment by arynyklas** [comment]
> @arynyklas For win7 have to change also deserialize:
> 
> ```
> if sys.platform == "win32":
>     def serialize(self, value: datetime.datetime):
>         return round((value - datetime.datetime(1970, 1, 1)).total_seconds())
>     def deserialize(self, value, parent=None):
>         return value
> else:
>     def serialize(self, value: datetime.datetime):
>         return round(value.timestamp())
>     def deserialize(self, value, parent=None):
>         return datetime.datetime.fromtimestamp(value)
> ```

deserialize need to return datetime.datetime

**Comment by fifthsound** [comment]
> @arynyklas For win7 have to change also deserialize:
> deserialize need to return datetime.datetime

Yes, I mean that the same error on windows also exist for deserialization.

Get this error: 
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\dispatcher\dispatcher.py", line 385, in start_polling
    allowed_updates=allowed_updates
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\bot\bot.py", line 111, in get_updates
    return [types.Update(**update) for update in result]
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\bot\bot.py", line 111, in <listcomp>
    return [types.Update(**update) for update in result]
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\types\base.py", line 95, in __init__
    self.props[key].set_value(self, value, parent=self)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\types\fields.py", line 58, in set_value
    value = self.deserialize(value, parent)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\types\fields.py", line 116, in deserialize
    return self.base_object.to_object(conf={'parent': parent}, data=value)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\types\base.py", line 153, in to_object
    return cls(conf=conf, **data)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\types\base.py", line 95, in __init__
    self.props[key].set_value(self, value, parent=self)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\types\fields.py", line 58, in set_value
    value = self.deserialize(value, parent)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\types\fields.py", line 116, in deserialize
    return self.base_object.to_object(conf={'parent': parent}, data=value)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\types\chat_member.py", line 97, in to_object
    return cls.resolve(conf=conf, **data)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\types\chat_member.py", line 88, in resolve
    return class_(**kwargs)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\types\base.py", line 95, in __init__
    self.props[key].set_value(self, value, parent=self)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\types\fields.py", line 58, in set_value
    value = self.deserialize(value, parent)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python37\lib\site-packages\
aiogram\types\fields.py", line 176, in deserialize
    return datetime.datetime.fromtimestamp(value)
OSError: [Errno 22] Invalid argument

**Pull request merged**